oxidation/libparsec/crates/crypto/src/sodiumoxide/sign.rs

```diff
@@ -113,14 +113,10 @@ impl VerifyKey {
     /// Verify a signature using the given [VerifyKey], `signature` and `message`
     pub fn verify_with_signature(
         &self,
-        raw_signature: &[u8],
+        raw_signature: [u8; Signature::BYTE_SIZE],
         message: &[u8],
     ) -> Result<Vec<u8>, CryptoError> {
-        if raw_signature.len() != Signature::BYTE_SIZE {
-            return Err(CryptoError::Signature);
-        }
-        let signature = Signature::try_from(raw_signature)
-            .expect("Precondition already checked for the signature size");
+        let signature = Signature::from_bytes(&raw_signature).expect("Unreachable");
```

Comment:
expect can occur if raw_signature is an invalid key, return a `Err(CryptoError::Signature)` instead